qutebrowser/config/config.py

```diff
@@ -162,13 +163,39 @@ def get_bindings_for(self, mode: str) -> Dict[keyutils.KeySequence, str]:
                 bindings[key] = binding
         return bindings
 
+    def _implied_cmd(self, cmdline: str) -> str:
+        """Return cmdline, or the implied cmd if cmdline is a set-cmd-text."""
+        try:
+            results = runners.CommandParser().parse_all(cmdline, aliases=False)
+            if len(results) == 0:
+                return ""
+            result = results[0]
+        except cmdexc.NoSuchCommandError:
+            return ""
+        if result.cmd.name != "set-cmd-text":
+            return cmdline
+        *flags, cmd = result.args
+        if "-a" in flags or "--append" in flags or not cmd.startswith(":"):
+            return ""  # doesn't look like this sets a command
+        return cmd.lstrip(":")
+
     def get_reverse_bindings_for(self, mode: str) -> '_ReverseBindings':
-        """Get a dict of commands to a list of bindings for the mode."""
+        """Get a dict of commands to a list of bindings for the mode.
+
+        This is intented for user-facing display of keybindings.
+        As such, bindings for 'set-cmd-text [flags] :<cmd> ...' are translated
+        to '<cmd> ...', as from the user's perspective these keys behave like
+        bindings for '<cmd>' (that allow for further input before running).
+
+        See #5942.
+        """
         cmd_to_keys: KeyConfig._ReverseBindings = {}
         bindings = self.get_bindings_for(mode)
         for seq, full_cmd in sorted(bindings.items()):
             for cmd in full_cmd.split(';;'):
```

Comment:
Wouldn't it make more sense to use `runners.CommandParser().parse_all(...)` here already? We could then probably pass the `ParseResult` to `self._implied_cmd` directly, and use `''.join(result.cmdline)` (I think? Not sure...) instead below.

Reply:
This does make sense, though changing the arg type to `runners.ParseResult` causes more circular import problems. I should probably just fix that